float64 error
float64 errorDerivative
